packages/core/TextSearch/TextSearchManager.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/*  text-searching controller */
+import BaseResult from './BaseResults'
+import { AnyConfigurationModel } from '../configuration/configurationSchema'
+
+import PluginManager from '../PluginManager'
+import QuickLRU from '../util/QuickLRU'
+import { searchType, BaseTextSearchAdapter } from '../data_adapters/BaseAdapter'
+import { readConfObject } from '../configuration'
+import TextSearchAdapterType from '../pluggableElementTypes/TextSearchAdapterType'
+
+export interface BaseArgs {
+  searchType: searchType
+  queryString: string
+  signal?: AbortSignal
+  limit?: number
+  pageNumber?: number
+}
+
+export interface Scope {
+  aggregate: boolean
```

Comment:
maybe rename this to `includeAggregateIndexes`

Reply:
Renamed to includeAggregateIndexes